File: leetcode2/1easy/최원준/Q3114.java
package Leetcode;

/*
1. 아이디어 :


2. 시간복잡도 :
O( )

3. 자료구조/알고리즘 :

 */

public class Q3114 {
    class Solution {
        public String findLatestTime(String s) {
            StringBuilder sb = new StringBuilder();
            if (s.charAt(0) == '?' && s.charAt(1) == '?') {
                sb.append("11");
            } else if (s.charAt(0) == '?') {
                if (Integer.parseInt((s.charAt(1))+"") >= 2) {
                    sb.append("0").append(s.charAt(1));
                } else {
                    sb.append("1").append(s.charAt(1));
                }
            } else if (s.charAt(1) == '?') {
                if (Integer.parseInt((s.charAt(0))+"") == 0) {
                    sb.append(s.charAt(0)).append("9");
                } else {
                    sb.append(s.charAt(0)).append("1");
                }
            } else {
                sb.append(s.charAt(0) + "" + s.charAt(1));
            }

            sb.append(":");

            if (s.charAt(3) == '?' && s.charAt(4) == '?') {
                sb.append("59");
            } else if (s.charAt(3) == '?') {
                sb.append("5").append(s.charAt(4));
            } else if (s.charAt(4) == '?') {
                sb.append(s.charAt(3)).append("9");
            } else {
                sb.append(s.charAt(3) + "" + s.charAt(4));
            }

            return sb.toString();
        }
    }
}

File: leetcode2/1easy/강병호/Q3114.py
class Leetcode.이재훈.Solution:
    def findLatestTime(self, s: str) -> str:
        # i - 0 : i - 1과 연관 / 3이상이면 무조건 0
        # i - 1 : i - 0과 연관 / 1이면 0~2
        time = list(s)

        t0, t1 = time[0], time[1]
        

        if t0 == '?':
            if t1 == '?':
                t0 = '1'
            else:
                t0 = '1' if int(t1) <= 1 else '0'
        
        time[0] = t0

        if t1 == '?':
            t1 = '1' if time[0] == '1' else '9'
        time[1] = t1

        t3, t4 = time[3], time[4]
        if t3 == '?':
            t3 = '5'
        time[3] = t3
        
        if t4 == '?':
            t4 = '9'
        time[4] = t4

        return ''.join(time)

        

